plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/guru/VMwareGuru.java

```diff
@@ -1104,13 +1124,22 @@ public List<Command> finalizeMigrate(VirtualMachine vm, Map<Volume, StoragePool>
             if (pool.getClusterId() != null) {
                 poolClusterId = pool.getClusterId();
             }
+            if (volume.getVolumeType().equals(Volume.Type.ROOT) && pool.isLocal()) {
+                isLocalStorageMigration = true;
```

Comment:
May be keep target VM pool and check if it is local for local migration (`StoragePool targetVmPool = null`)
```suggestion
            if (volume.getVolumeType().equals(Volume.Type.ROOT)) {
                targetVmPool = pool;
```